RELEASE.md

```diff
@@ -16,6 +16,7 @@
 
 ## Bug fixes and other changes
 * Consolidated dependencies and optional dependencies in `pyproject.toml`.
+* Pin `pip<23.2` due to a breaking change. See https://github.com/kedro-org/kedro/pull/2813
```

Comment:
Hopefully we will not have to release Kedro like this 😢 

Also, notice that we only changed the CI requirements, Kedro `pyproject.toml` is intact